kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -70,74 +72,77 @@ function _collectionState(data) {
-  return new Promise((resolve, reject) => {
-    _getChannelList()
-      .then((channelList) => {
-        if (channelList && channelList.length) {
-          const cookieCurrentChannelId = cookiejs.get('currentChannel');
-          if (channelList.some((channel) => channel.id === cookieCurrentChannelId)) {
-            currentChannelId = cookieCurrentChannelId;
-          } else {
-            currentChannelId = channelList[0].id;
-          }
-        }
-        resolve(currentChannelId);
-      });
-  });
+
+function _explorePageTitle(title) {
+  if (title) {
+    return `Explore - ${title}`;
+  }
+  return 'Explore';
+}
+
+function _learnPageTitle(title) {
+  if (title) {
+    return `Learn - ${title}`;
+  }
+  return 'Learn';
+}
+
+function _errorTitle(title) {
+  if (title) {
+    return `Error - ${title}`;
+  }
+  return 'Error';
 }
 
 
 /*
- * Returns a promise that fetches the the root topic id of the current channel.
+ * Returns the 'default' channel ID:
+ * - if there are channels and they match the cookie, return that
+ * - else if there are channels, return the first one
+ * - else return null
  */
-function _getCurrentChannelRootTopicId() {
-  let currentChannelRootTopicId = null;
-  return new ConditionalPromise((resolve, reject) => {
-    const currentChannelIdPromise = _getCurrentChannelId();
-    const channelListPromise = _getChannelList();
-    Promise.all([currentChannelIdPromise, channelListPromise])
-    .then(([currentChannelId, channelList]) => {
-      if (currentChannelId && channelList) {
-        for (const channel in channelList) {
-          if (channelList[channel].id === currentChannelId) {
-            currentChannelRootTopicId = channelList[channel].root_pk;
-            break;
-          }
-        }
-      }
-      resolve(currentChannelRootTopicId);
-    });
-  });
+function _getDefaultChannelId(store, channelList) {
+  if (channelList && channelList.length) {
+    const cookieVal = cookiejs.get('currentChannelId');
+    if (channelList.some((channel) => channel.id === cookieVal)) {
+      return cookieVal;
+    }
+    return channelList[0].id;
+  }
+  return null;
 }
 
-function _updateChannelList(store) {
-  const channelPromise = _getChannelList();
-  channelPromise.then((channelList) => {
-    store.dispatch('SET_CHANNEL_LIST', channelList);
-  });
-  return channelPromise;
+
+/*
+ * Set channel state info.
+ * Returns `true` if the requested channel ID is available in the list.
+ */
+function _handleChannels(store, currentChannelId, channelList) {
+  store.dispatch('SET_CHANNEL_LIST', channelList);
+  store.dispatch('SET_CURRENT_CHANNEL', currentChannelId);
+  ContentNodeResource.setChannel(currentChannelId);
+  cookiejs.set('currentChannelId', currentChannelId);
+
+  if (!channelList.find(channel => channel.id === currentChannelId)) {
+    store.dispatch('CORE_SET_ERROR', 'Channel not found');
+    store.dispatch('CORE_SET_PAGE_LOADING', false);
+    document.title = _errorTitle();
```

Comment:
I have been trying to nail down what bothers me about this, and I think it's the fact that this is not reactive, and so the document title is not represented in state.

I think we should do this by having a core-title plugin that can be passed in the right title based on Vuex state (this would also allows us to move the title names into Vue components and solve our $trs issue simply).